Installation issue: libheif (trying to use system gdk-pixbuf)

### Steps to reproduce the issue

```console
$ spack --env zen2-gcc13 find -lvcdf libheif     
...
-- linux-rhel8-zen2 / %c,cxx=gcc@13.4.0 -------------------------
 -   waf26e5 libheif@1.12.0+ipo build_system=cmake build_type=RelWithDebInfo generator=make
[+]  eufn52p     cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=RelWithDebInfo
[+]  gu5vdnc         curl@8.15.0+gssapi~ldap+libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=mbedtls,openssl
[+]  r2dc5ck             krb5@1.21.3+shared build_system=autotools
[+]  kdrpmhr                 bison@3.8.2~color build_system=autotools
[+]  vlbkf4y                     m4@1.4.20+sigsegv build_system=autotools
[+]  focymqv                         libsigsegv@2.14 build_system=autotools
[+]  iartorz                 diffutils@3.12 build_system=autotools
[+]  zfcjhbz                     libiconv@1.18 build_system=autotools libs:=shared,static
[+]  zirylds                 findutils@4.10.0 build_system=autotools patches:=440b954
[+]  ful6i3c                 gettext@0.23.1+bzip2+curses+git+libunistring+libxml2+pic+shared+tar+xz build_system=autotools
[+]  btav5ox                     libxml2@2.13.5+http+pic~python+shared build_system=autotools
[+]  ahyoscx                     tar@1.35 build_system=autotools zip=pigz
[+]  vxpii7z                         pigz@2.8 build_system=makefile
[+]  n64duty                         zstd@1.5.7+programs build_system=makefile compression:=lz4,lzma,zlib libs:=shared,static
[+]  3367tmt                             lz4@1.10.0+pic build_system=makefile libs:=shared,static
[+]  ihgv3uq                 libedit@3.1-20240808 build_system=autotools
[+]  xtgeq4a             libidn2@2.3.7 build_system=autotools
[+]  xbdx6lg                 libunistring@1.2 build_system=autotools
[+]  24bzzhf             libssh2@1.11.1+shared build_system=autotools crypto=mbedtls
[+]  pyx73rp                 xz@5.6.3+pic build_system=autotools libs:=shared,static
[+]  crudhjn             mbedtls@2.28.9+pic build_system=makefile build_type=RelWithDebInfo libs:=shared,static
[+]  7wi62dj             nghttp2@1.67.1 build_system=autotools
[+]  unrffxy             openssl@3.6.0~docs+shared build_system=generic certs=system
[+]  edlxjs4             perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic
[+]  pxa3pju                 berkeley-db@18.1.40+cxx+docs+stl build_system=autotools patches:=26090f4
[+]  jiyqkwo                 bzip2@1.0.8~debug+pic+shared build_system=generic
[+]  m4ifnmh                 gdbm@1.25 build_system=autotools
[+]  f375v3a                     readline@8.3 build_system=autotools patches:=21f0a03
[+]  2pp7u3i             pkgconf@2.5.1 build_system=autotools
[+]  vet42cx         ncurses@6.5-20250705+symlinks+termlib abi=5 build_system=autotools patches:=7a351bc
[+]  p4i3vud         zlib@1.3.1+optimize+pic+shared build_system=makefile
[+]  c4lvqa5     compiler-wrapper@1.0 build_system=generic
[e]  gw3zygo     gcc@13.4.0+binutils~bootstrap+graphite~mold+nvptx~piclibs~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran,lto'
[+]  gd56l37     gcc-runtime@13.4.0 build_system=generic
[e]  3kzdcra     glibc@2.28 build_system=autotools
[+]  ymb7dyf     gmake@4.4.1~guile build_system=generic
...
```

Then running
```console
$ spack --env zen2-gcc13 install libheif
```
fals with an error about failing to copy a file in /usr/lib64/gdk-pixbuf

### Error message

<details><summary>Error message</summary>

<pre>
...
[ 90%] Built target pixbufloader-heif
[100%] Built target heif-thumbnailer
make[1]: Leaving directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/spack-build-waf26e5'
/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/bin/cmake -E cmake_progress_start /tmp/swinst26/spackswinst26/spack-stage/spack-stage-libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/spack-build-waf26e5/CMakeFiles 0
/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25/bin/make  -f CMakeFiles/Makefile2 preinstall
make[1]: Entering directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/spack-build-waf26e5'
make[1]: Nothing to be done for 'preinstall'.
make[1]: Leaving directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/spack-build-waf26e5'
Install the project...
/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/bin/cmake -P cmake_install.cmake
-- Install configuration: "RelWithDebInfo"
-- Installing: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/lib64/pkgconfig/libheif.pc
-- Installing: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/bin/heif-convert
...
-- Installing: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/lib64/cmake/libheif/libheif-config.cmake
-- Installing: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/lib64/cmake/libheif/libheif-config-relwithdebinfo.cmake
-- Installing: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/lib64/cmake/libheif/libheif-config-version.cmake
-- Installing: /usr/lib64/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-heif.so
CMake Error at gdk-pixbuf/cmake_install.cmake:60 (file):
  file INSTALL cannot copy file
  "/tmp/swinst26/spackswinst26/spack-stage/spack-stage-libheif-1.12.0-waf26e5h42h3dl4rwfsezrwtiti2lasx/spack-build-waf26e5/gdk-pixbuf/libpixbufloader-heif.so"
  to "/usr/lib64/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-heif.so":
  Permission denied.
Call Stack (most recent call first):
  cmake_install.cmake:53 (include)


make: *** [Makefile:103: install] Error 1
</pre></details>

The spack info libheif does not list gdk-pixbuf as dependency, but it clearly depends on such and is picking up the system installed version, tries to add a library under the gdk-pixmap install directory, and fails due to the lack of access.

While gdk-pixbuf is not installed through Spack in this environment yet, at least one build of gdk-pixbuf is concretized in the environment (spec included with the files below).

### Information on your system

$ spack debug report
* **Spack:** 1.1.0
* **Builtin repo:** https://github.com/spack/spack-packages/commit/119680aeee8ea802c6111b7167583bddef97e82f
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen2



### Additional information

REquested spack files:
- [spack-build-env.txt](https://github.com/user-attachments/files/26764975/spack-build-env.txt)
- [spack-build-out.txt](https://github.com/user-attachments/files/26764974/spack-build-out.txt)

Other spack buiild files:
[spack-build-01-cmake-out.txt](https://github.com/user-attachments/files/26764989/spack-build-01-cmake-out.txt)
[spack-build-02-build-out.txt](https://github.com/user-attachments/files/26764992/spack-build-02-build-out.txt)
[spack-build-03-install-out.txt](https://github.com/user-attachments/files/26764991/spack-build-03-install-out.txt)
[spack-build-env-mods.txt](https://github.com/user-attachments/files/26764990/spack-build-env-mods.txt)

Specs of pakcages:
- [Full spec of libheif trying to install](https://github.com/user-attachments/files/26764980/libheif.spec-full.txt)
- [Specs of ](https://github.com/user-attachments/files/26764982/gdk-pixbuf.spec-full.txt)

No maintainers


### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/libheif/package.py
+++ b/repos/spack_repo/builtin/packages/libheif/package.py
@@ -15,9 +15,21 @@ class Libheif(CMakePackage):
 
     license("LGPL-3.0-or-later")
 
+    version("1.21.2", sha256="79996de959d28ca82ef070c382304683f5cdaf04cbe2953a74587160a3710a36")
     version("1.12.0", sha256="086145b0d990182a033b0011caadb1b642da84f39ab83aa66d005610650b3c65")
+
+    variant("gdk-plugin", default=True, description="Build gdk-pixbuf plugin")
 
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
     depends_on("cmake@3.13:", type="build")
+    depends_on("gdk-pixbuf", when="+gdk-plugin")
+    depends_on("jpeg")
+    depends_on("libpng")
+
+    def cmake_args(self):
+        args = [
+                self.define_from_variant("WITH_GDK_PIXBUF", "gdk-plugin"),
+        ]
+        return args

--- a/repos/spack_repo/builtin/packages/libheif/package.py
+++ b/repos/spack_repo/builtin/packages/libheif/package.py
@@ -30,6 +30,6 @@ class Libheif(CMakePackage):
 
     def cmake_args(self):
         args = [
-                self.define_from_variant("WITH_GDK_PIXBUF", "gdk-plugin"),
+            self.define_from_variant("WITH_GDK_PIXBUF", "gdk-plugin"),
         ]
         return args

